src/cmd/cli/command/hint.go

```diff
@@ -23,10 +23,19 @@ func prettyExecutable(def string) string {
 	if err != nil {
 		return def
 	}
+
+	// for npm/npx defang is executed within a child process,
+	// but we want to use parent process command line
+	execLine := os.Getenv("DEFANG_COMMAND_EXECUTOR")
```

Comment:
DEFANG_COMMAND_EXECUTOR is the key value passed in from the cli.js used in npx calls. This will no occur normally. Will contain something like "npx defang"

Reply:
add tests pls